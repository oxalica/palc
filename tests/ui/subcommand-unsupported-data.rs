use palc::Subcommand;

#[derive(Subcommand)]
struct Unit;

#[derive(Subcommand)]
struct Tuple();

#[derive(Subcommand)]
struct Named {
    a: (),
}

#[derive(Subcommand)]
union Union {
    a: (),
}

struct AssertImplSubcommand
where
    Unit: Subcommand,
    Tuple: Subcommand,
    Named: Subcommand,
    Enum: Subcommand;

fn main() {}
