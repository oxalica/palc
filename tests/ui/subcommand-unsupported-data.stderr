error: structs are only supported by `derive(Args)`, not by `derive(Subcommand)`
 --> tests/ui/subcommand-unsupported-data.rs:3:10
  |
3 | #[derive(Subcommand)]
  |          ^^^^^^^^^^
  |
  = note: this error originates in the derive macro `Subcommand` (in Nightly builds, run with -Z macro-backtrace for more info)

error: structs are only supported by `derive(Args)`, not by `derive(Subcommand)`
 --> tests/ui/subcommand-unsupported-data.rs:6:10
  |
6 | #[derive(Subcommand)]
  |          ^^^^^^^^^^
  |
  = note: this error originates in the derive macro `Subcommand` (in Nightly builds, run with -Z macro-backtrace for more info)

error: structs are only supported by `derive(Args)`, not by `derive(Subcommand)`
 --> tests/ui/subcommand-unsupported-data.rs:9:10
  |
9 | #[derive(Subcommand)]
  |          ^^^^^^^^^^
  |
  = note: this error originates in the derive macro `Subcommand` (in Nightly builds, run with -Z macro-backtrace for more info)

error: only enums are supported
  --> tests/ui/subcommand-unsupported-data.rs:14:10
   |
14 | #[derive(Subcommand)]
   |          ^^^^^^^^^^
   |
   = note: this error originates in the derive macro `Subcommand` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0412]: cannot find type `Enum` in this scope
  --> tests/ui/subcommand-unsupported-data.rs:24:5
   |
24 |     Enum: Subcommand;
   |     ^^^^ not found in this scope
